docs/src/main/asciidoc/se/openapi/openapi-generator.adoc

```diff
@@ -23,9 +23,277 @@
 :rootdir: {docdir}/../..
 
 include::{rootdir}/includes/se.adoc[]
+:snippets: {sourcedir}/se/openapi/OpenApiGeneratorSnippets.java
 
-== Coming Soon
+:helidon-client-xref: {webclient-page}
 
-The support for link:{openapi-generator-tool-site-url}[OpenAPI generator] is not available yet.
+include::{rootdir}/includes/openapi/openapi-generator.adoc[tag=preamble]
 
-You can track the progress for this feature here: https://github.com/helidon-io/helidon/issues/7943.
+include::{rootdir}/includes/openapi/openapi-generator.adoc[tags=intro;coords;config;usage-basics;usage-grouping-intro;usage-grouping-server]
+
+The generator creates Helidon routing logic based on the longest common path prefix shared among the operations that are grouped into each API.
+
+[NOTE]
+====
+If the operations in an API have no common prefix then the generated routing will be inefficient at runtime.
+The generator logs a warning and includes a `TODO` comment in the generated routing.
+
+Review the paths and the `tags` settings in your OpenAPI document and consider revising one or the other so all operations in each API share a common path prefix. If you do not have control over the OpenAPI document or do not want to change it, consider specifying the generator option `x-helidon-groupBy first-path-segment` which groups operations into APIs not by `tags` value but by the first segment of each operation's path.
+====
+
+include::{rootdir}/includes/openapi/openapi-generator.adoc[tags=usage-grouping-client;usage-after-grouping;using-generated-code-intro;using-generated-code-server-intro]
+
+If you choose to generate interfaces for the APIs, the generator creates routing rules for the API services it generates but you write virtually all of the logic to process incoming requests by implementing the very short methods generated in the implementation class.
+
+The rest of this sections focuses on your next steps if, on the other hand, you decide to generate abstract classes.
+
+==== What you _must_ do: implement your business logic and send the response
+The generator creates an implementation class as well as the abstract class for each API.
+The implementation class contains a `handle` method for each API operation with a very simple method body that returns a not-yet-implemented HTTP status in the response. The following example shows the generated method for the `addPet` OpenAPI operation.
+[source,java]
+.The generated `handleAddPet` method in the `PetApiImpl` class
+----
+include::{snippets}/[tags=PetServiceImplCustomclass-header;PetServiceImplCustomhandleAddPetDecl;PetServiceImplCustomhandleAddPetGenerated;PetServiceImplCustomhandleAddPetEnd;PetServiceImplCustomclass-end, indent=0]
```

Comment:
`include::{sourcedir}/se/openapi/OpenApiGeneratorSnippets.java`

---

Is it possible to use whole snippets instead of composing them with multiple small ones ?

I know you are trying to avoid repetition, however I believe it's better to repeat a few lines in the java files, and have each snippets visible as a whole.

Reply:
Repeating code to have "self-contained" snippet fragments inflates the snippets file  by 30% (388 lines to 501). 

I know file counts and sizes are also a concern of yours, and as you noted repetition is one of mine due to the potential maintenance issues if and when we need to revise the snippets. 

I also get your point about having self-contained snippets; to me preserving the natural structure of a single copy of the code is very beneficial even if punctuated with `// tag` and `// end` comments. I think at least in this case the change would be a net negative for us.

Reply:
I did the [exercise](https://gist.github.com/romain-grecourt/f6d92cd092a27b49e5f2834c59e1f150) and ended up with 425 lines, that includes the instructions at the top that take 26 lines (made some updates).

This is comparable to [HcpSnippets](https://github.com/helidon-io/helidon/blob/a07579b7635f45d712066230e39b9727d8e86984/docs/src/main/java/io/helidon/docs/mp/integrations/HcvSnippets.java#L421) which is 421 lines long.

I strictly followed the current pattern and ended-up with something comparable to an existing snippets class.

---

I value simplicity in our asciidoc files, ideally they should look less like code and more like text.

Reply:
Did your snippets file compile for you? It failed for me.

Clearly we disagree on what's a better approach for structuring the snippets file. Nevertheless, I have taken your changes, fixed the compilation error, and will push the commit momentarily.